File: backend/OperationHistory.BLL/DataTransferObjects/OperationDto.cs
using Common.Enum;

namespace OperationHistory.BLL.DataTransferObjects;

public class OperationDto {
    public Guid Id { get; set; }
    public Guid AccountId { get; set; }
    public Guid? LoanId { get; set; }
    public OperationType Type { get; set; }
    public OperationStatus Status { get; set; }
    public int Amount { get; set; }
    public string? Message { get; set; }
    
}

File: frontend/web-app-employee/src/services/operationHistory/models/OperationDto.ts


export enum OperationReason {
    Cash = "Cash",
    Loan = "Loan",
}

export type OperationDto = {
    id: string
    accountId: string
    loanId?: string
    type: OperationType
    status: OperationStatus
    reason: OperationReason
    amount: number
    createdAt: string
    message?: string
    currencyType: string
}

export enum OperationType {
    Deposit = "Deposit",
    Withdraw = "Withdraw",
}

export enum OperationStatus {
    Success = "Success",
    Failure = "Failure",
    Processing = "Processing",
}